archive/tar.go

```diff
@@ -11,6 +11,82 @@ import (
 	"time"
 )
 
+func WriteFilesToTar(dest string, uid, gid int, files ...string) (string, map[string]struct{}, error) {
+	hasher := sha256.New()
+	f, err := os.Create(dest)
+	if err != nil {
+		return "", nil, err
+	}
+	w := io.MultiWriter(hasher, f)
+	tw := tar.NewWriter(w)
+
+	fileSet := map[string]struct{}{}
+	for _, file := range files {
+		if AddFileToArchive(tw, file, uid, gid, fileSet) != nil {
+			return "", nil, err
+		}
+	}
+	_ = tw.Close()
+	defer f.Close()
```

Comment:
This defer `f.Close()` should move directly below the error check after `os.Create`, so the file still gets closed even if the function exits with an error on line 25.